utils/pstor/src/legacy.rs

```diff
@@ -0,0 +1,49 @@
+use crate::{key_prefix_obj, Error, StorableObjectType, Store};
+use tracing::info;
+
+/// Returns the legacy key prefix that was used for the keys, when running from within the cluster.
+fn legacy_key_prefix() -> String {
```

Comment:
suggest having legacy module and within legacy have another file called product_v1 ?
Then don't add legacy prefix to all of these since we're already in legacy?